source/extensions/filters/http/grpc_field_extraction/filter.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/extensions/filters/http/grpc_field_extraction/v3/config.pb.h"
+#include "envoy/extensions/filters/http/grpc_field_extraction/v3/config.pb.validate.h"
+#include "envoy/http/filter.h"
+
+#include "source/extensions/filters/http/common/factory_base.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+#include "source/extensions/filters/http/grpc_field_extraction/filter_config.h"
+#include "source/extensions/filters/http/grpc_field_extraction/message_converter/message_converter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+
+inline constexpr const char kFilterName[] = "envoy.filters.http.grpc_field_extraction";
+
+class Filter : public Envoy::Http::PassThroughDecoderFilter,
+               Envoy::Logger::Loggable<Envoy::Logger::Id::filter> {
+public:
+  Filter(const FilterConfig& config) : filter_config_(config) {}
+
+  Envoy::Http::FilterDataStatus decodeData(Envoy::Buffer::Instance& data, bool end_stream) override;
+
+  Envoy::Http::FilterHeadersStatus decodeHeaders(Envoy::Http::RequestHeaderMap& headers,
+                                                 bool end_stream) override;
+
+private:
+  struct HandleDecodeDataStatus {
+    HandleDecodeDataStatus() : got_messages(true) {}
```

Comment:
nit: I prefer you remove this default constructor that sets `got_messages = true`. Otherwise it's not obvious in the caller code if this indicates success or what is being returned:

```cpp
return {};
```

Prefer explicitly spelling it out for readability:

```cpp
return HandleDecodeDataStatus{
  .got_messages = true
}
```

or 

```cpp
return HandleDecodeDataStatus{/*got_messages=*/ true};
```

Reply:
IMHO, the current approach is more NIT.  It is straightforward when decodeDataStruct isn't set, it is OK and got_messages=true by default.